src/timeseriesflattener/flattened_dataset.py

```diff
@@ -402,124 +380,54 @@ def _flatten_temporal_values_to_df(  # noqa pylint: disable=too-many-locals
 
         # handle embedding and dimensionality reduction if text predictor
         if isinstance(output_spec, TextPredictorSpec):
-            df = TimeseriesFlattener._embed_text_values(
+            df = embed_text_column(
                 df=df,
+                text_col_name="value",
                 embedding_fn=output_spec.embedding_fn,
-                dim_reduction_fn=output_spec.dim_reduction_fn,
+                embedding_fn_kwargs=output_spec.embedding_fn_kwargs,
             )
 
         # If resolve_multiple generates empty values,
         # e.g. when there is only one prediction_time within look_ahead window for slope calculation,
         # replace with NaN
 
         # Rename column
-        df = TimeseriesFlattener._rename_value_column(df=df, output_spec=output_spec)
+        df = MultiIndexHandler.rename_value_column(df=df, output_spec=output_spec)
 
         # Find value_cols and add fallback to them
-        value_col_str_name = TimeseriesFlattener._get_value_col_str_name(
+        value_col_str_name = MultiIndexHandler.get_value_col_str_name(
             df=df,
             output_spec=output_spec,
         )
-        df = TimeseriesFlattener._add_fallback_to_value_cols(
+        df = MultiIndexHandler.add_fallback_to_value_cols(
```

Comment:
```suggestion
        df = MultiIndexHandler.replace_na_in_value_col_with_fallback(
```

I assume this is what the function does? :-)